napari/layers/shapes/_shapes_mouse_bindings.py

```diff
@@ -100,8 +136,20 @@ def select(layer, event):
         layer._update_thumbnail()
 
 
-def add_line(layer, event):
-    """Add a line."""
+def add_line(layer: Shapes, event: ReadOnlyWrapper) -> None:
+    """Add a line.
```

Comment:
The edge width is purely a visual which is documented elsewhere. I added to the docstring, it basically starts with 2 points in data space where the second point has a minor offset to the first one if mouse has not moved yet. When the mouse starts moving the second point will track the cursor. Not certain whether this offset is required as someone could misclick and then just release. Now this would create a tiny line while I personally would expect that this would delete the line just as when you have a polygon with less than 3 vertices. This would be something to for an issue / separate PR though.

Reply:
AFAIK The offset is requires because otherwise you'd get a degenerate 0-length line when you start drawing (since the line is constantly updated to the last position) and this can mess up the math later.

Reply:
How would a 0-length line be different from the situation of starting to draw a polygon? A user clicks and 2 points are being created, 1 starting point and one active point which before moving the mouse is at exactly the same coordinate.
 
If these have the same coordinates there should be no problem while `layer._is_creating` is `True`. If finishing drawing a line like this should just be deleted as the user probably initiated the draw by mistake.

Reply:
@melonora could you make a new issue? In the meantime this can be resolved, although I'd prefer if the docstring was wrapped to 80c like the rest of the code. 😅